### ISSUE TITLE
[Bug]: Honkai Impact 3rd • Anti-Cheat Error

### ISSUE THREAD

**Issue by exterpolation** [CollapseLauncher/Collapse#357]
### Affected Version

Collapse 1.72.14-PRE and Previous

### Bug Behavior

[DxDiag.txt](https://github.com/CollapseLauncher/Collapse/files/13847429/DxDiag.txt)
[log-2024-01-06-pre1.72.14-id1.log](https://github.com/CollapseLauncher/Collapse/files/13847450/log-2024-01-06-pre1.72.14-id1.log)

Once the game it launches, the game freezes for a split second in the "Warning" Screen, then the game closes itself and throws a "Warning (13, 131083, 140)" Window, the installation folder was the same as the official launcher, where it behaves normally.

### Expected Behavior

To run the game as if it was launcher through the official launcher.

### Steps to reproduce

[1] Switch to Honkai Impact 3d (Global)
[2] Launch the game 

### Screenshot(s)

[Image of the Warning Window](https://cdn.discordapp.com/attachments/1116163954071318658/1192875862991843388/image.png?ex=65aaab38&is=65983638&hm=d9315d04857412d0c00e891af4a52dbd5368e9e9ad10c06b455da198a28b2660&)

**Comment by Cryotechnic** [comment]
This can sometimes be caused by the "Core Isolation" feature being enabled in Windows Security, which causes the anti-cheat to fail to initialize properly. Please try to disable "Core Isolation" and see if it resolves the issue. 

Should that not solve the issue, please make sure that you have a stable internet connection and, if using a VPN, that it is connected to a node in a supported country for the region you are trying to connect to.

Do let us know if that resolves the issue, thanks for using Collapse!

**Comment by exterpolation** [comment]
I've tried turning off Core Isolation (Checked and it was already off) and it was still the same

As for the connection I have a stable ethernet port (200mbps) without using any VPNs, didnt really change much

**Comment by bagusnl** [comment]
can you try launching the game by running the BH3.exe file directly once and go to the main screen then close it and try again using Collapse?

**Comment by bagusnl** [comment]
Oh, I have noticed this on your dxdiag log
```
------------
DxDiag Notes
------------
      Display Tab 1: The file C:\Windows\System32\DriverStore\FileRepository\nvaeig.inf_amd64_ae0e4b8fa6a17cef\nvldumdx.dll,C:\Windows\System32\DriverStore\FileRepository\nvaeig.inf_amd64_ae0e4b8fa6a17cef\nvldumdx.dll,C:\Windows\System32\DriverStore\FileRepository\nvaeig.inf_amd64_ae0e4b8fa6a17cef\nvldumdx.dll,C:\Windows\System32\DriverStore\FileRepository\nvaeig.inf_amd64_ae0e4b8fa6a17cef\nvldumdx.dll is not digitally signed, which means that it has not been tested by Microsoft's Windows Hardware Quality Labs (WHQL).  You may be able to get a WHQL logo'd driver from the hardware manufacturer.
      Display Tab 2: The file C:\Windows\System32\DriverStore\FileRepository\nvaeig.inf_amd64_ae0e4b8fa6a17cef\nvldumdx.dll,C:\Windows\System32\DriverStore\FileRepository\nvaeig.inf_amd64_ae0e4b8fa6a17cef\nvldumdx.dll,C:\Windows\System32\DriverStore\FileRepository\nvaeig.inf_amd64_ae0e4b8fa6a17cef\nvldumdx.dll,C:\Windows\System32\DriverStore\FileRepository\nvaeig.inf_amd64_ae0e4b8fa6a17cef\nvldumdx.dll is not digitally signed, which means that it has not been tested by Microsoft's Windows Hardware Quality Labs (WHQL).  You may be able to get a WHQL logo'd driver from the hardware manufacturer.
      ```
      
      You might want to reinstall your GPU driver by firstly uninstalling it completely using DDU.
      https://www.wagnardsoft.com/content/Display-Driver-Uninstaller-DDU-V18070-Released

**Comment by exterpolation** [comment]
I'll try it out soon, but why does it work on the official launcher instead of collapse?

**Comment by exterpolation** [comment]
By the way, they're not signed as i used NVTrimmer to remove telemetry and useless features from NVIDIAs latest drivers

**Comment by exterpolation** [comment]
Tried DDU, nothing really changed

**Comment by bagusnl** [comment]
can you try to run the Collapse from this build and test it out?
https://github.com/CollapseLauncher/Collapse/actions/runs/7269824437
This one is old build, so don't replace your current installation with it, just extract it anywhere and launch CollapseLauncher.exe and try if it works

**Comment by exterpolation** [comment]
Nope, same error, same behaviour

**Comment by bagusnl** [comment]
This is quite weird because that commit has the old way of running the games which might ignores some rouge executables on the game directory.

Can you do this command in CMD on the directory of your HI3 install?
`tree /F /A >HI3_Tree.txt`

Then upload the `HI3_Tree.txt` file here

**Comment by exterpolation** [comment]
[HI3_Tree.txt](https://github.com/CollapseLauncher/Collapse/files/13853650/HI3_Tree.txt)

Here you go, done it

**Comment by bagusnl** [comment]
![Tree_Guide](https://github.com/CollapseLauncher/Collapse/assets/28079733/434fe26d-4038-435e-9276-c5922c73354c)
Sorry for not being clear on the guide to get the file tree, here hopely is

**Comment by exterpolation** [comment]
[HI3_Tree.txt](https://github.com/CollapseLauncher/Collapse/files/13853795/HI3_Tree.txt)

Here's the actual one, lol i shouldve actually done "cd 'honkai path folder'" before running tree

**Comment by BobbyWibowo** [comment]
I'm also getting this today, but it also happens even when I launch the game normally with the game's original launcher. Even with Collapse Launcher confirmed not running in the background.

Also, unlike OP's implication that the anti-cheat error would trigger immediately, it'd only happen to me about 5 minutes after the game launched. Sorta implying that it'd only trigger awhile after the game's anti-cheat has found something in my files, as opposed to monitoring running background processes.

**Comment by BobbyWibowo** [comment]
Can confirm that on my case this had nothing to do with Collapse Launcher.

As I did some investigation, I found this random reddit thread, https://www.reddit.com/r/houkai3rd/comments/rj058b/comment/hp7728e/. Though their case was a bit more extreme since it caused a hard OS crash instead. But anyways, the OP speculated that it had something to do with USB erroring out, and Honkai's AC in particular not liking that.

That train of thought got me thinking as to whether anything was different with the state of my plugged USB devices on that particular moment, and I found that I did have a USB bluetooth dongle connected for a pair of Logitech keyboard+mouse, that happened to turned-off. I disconnected it and rebooted my PC, and the game continued to run normally ever since, even when launched via Collapse Launcher.

Perhaps the fact that the keyboard+mouse were turned-off, caused the dongle to continuously ping or whatever, and was then logging errors and whatever else? And maybe Honkai's AC just didn't like that? Unsure. It'll forever remain a mystery lmao

**Comment by bagusnl** [comment]
> [HI3_Tree.txt](https://github.com/CollapseLauncher/Collapse/files/13853795/HI3_Tree.txt)
> 
> Here's the actual one, lol i shouldve actually done "cd 'honkai path folder'" before running tree

There doesn't seem to be any rouge dll/exe around so your installation is fine. 

Though as @BobbyWibowo stated, can you try running the game with minimal USB devices plugged in (only the necessary ones to run the game like mouse+keyboard, then restart the system)  ?

**Comment by exterpolation** [comment]
Tried it, yet nothing changed again. Weird stuff since the official launcher works as normally.

Another thing that i have noticed (which is minor) i can tell that the game is not gonna launch because the custom cursor you have inside honkai wont load, while when the game launcher perfectly it does load

**Comment by exterpolation** [comment]
> Is it possible that you both have a different anti-cheat interfering with Honkai Impact's one? For example, ~~from my time lurking in the Honkai Impact 3rd Discord's technical-help channel,~~ I've [heard (please join the HI3 Discord server to see this message)](https://discord.com/channels/418652140454674432/427673685026013184/1167310336945311744) that Riot Vanguard may not allow HI3 to initialize their anti-cheat.
> 
> Try closing every single program you can in the system tray (the bottom right of your task bar) and/or every non-essential process via the Task Manager and see if that fixes the issue.

Now that i think about it, I do have Valorant installed, alas Vanguard running. I'll try your fix once i'm home, thanks btw

**Comment by shatyuka** [comment]
@exterpolation Is the issue resolved after turning off Vanguard anti-cheat?

**Comment by exterpolation** [comment]
Forgot to reply, but no, nothing has changed

**Comment by shatyuka** [comment]
Try running the launcher with "Run as administrator".
If it still not work, I want some additional information of your system settings:
Result of command `whoami /groups`.
And group policy(gpedit.msc), "Computer Configuration"->"Windows Settings"->"Security Settings"->"Local Policies"->"Security Options". Have a screenshot of all entries start with "User Account Control".

**Comment by shatyuka** [comment]
can't reproduce

**Comment by Gotyourdetailz** [comment]
Even tho im very to late to this, I just had the same issue for ZZZ and thought I'd help anyone having the same issue as I am. I installed revi os which is a custom os based on improving latency and performance, after installing it he has a dedicated app for certain tweaks you might want, One of which is called "User Account Control" this can be either turned off or on, and if turned off it will elevate everything that you run to admin elevation without asking first, but if turned on it'll give you standard user privileges, like normal. You want to leave this on, if turned off you'll be thrown an anti cheat error. I'll attach photos of the error I got and the setting I'm talking about.

Btw having admin privileges on for everything is bad practice anyway, and a lot of applications/installers are built to be ran with standard permissions unless told otherwise. You'll save yourself a lot of headaches knowing this.

Hope this helps!
![Medal_w6gC57fghh](https://github.com/user-attachments/assets/7d88e96b-608e-4812-aab7-4e778b0681b5)
![sqLTHWOl1w](https://github.com/user-attachments/assets/6951f45c-ee7b-4490-83f3-921b5f3c4315)
